Flaky integration tests

I have been seeing randomly difficult to reproduce errors in the integration tests when ran in Github. Simply rerunning them makes the execution successful at some point.

These are some of the failed executions:

- [Link](https://github.com/kubearchive/kubearchive/actions/runs/12753717166/job/35545977190) - Timeout on log retrieval:

```
panic: test timed out after 10m0s
	running tests:
		TestArchiveAndRead (9m53s)
```

- [Link](https://github.com/kubearchive/kubearchive/actions/runs/12754142764/job/35547320965) - Timeout reached on Pagination test 
- [Link](https://github.com/kubearchive/kubearchive/actions/runs/12754142764/job/35574838237) - Timeout reached on knative-eventing rollout
- [Link](https://kubernetes.io/docs/reference/kubectl/quick-reference/#viewing-and-finding-resources) - api port-forwarding fails

```
E0115 16:20:09.413446   24728 portforward.go:424] "Unhandled Error" err="an error occurred forwarding 8081 -> 8081: error forwarding port 8081 to pod 8fef7bf251e68dc5295ab524b3fdaaf1174580536ce45fac3a12ac7f709e85c9, uid : network namespace for sandbox \"8fef7bf251e68dc5295ab524b3fdaaf1174580536ce45fac3a12ac7f709e85c9\" is closed" logger="UnhandledError"
``` 


It seems that the first issue happens more often so it should be addressed first by extending the timeout

EDIT: updated some links (old attempts can be accessed from Github, they are not overriden)

--- a/.github/workflows/go_test_integration.yml
+++ b/.github/workflows/go_test_integration.yml
@@ -57,4 +57,4 @@ jobs:
       - name: Test Go code
         run: |
           export KO_DOCKER_REPO="kind.local"
-          go test -count=1 -v ./test/integration -tags=integration
+          go test -count=1 -v ./test/integration -tags=integration -timeout 30m

--- a/cmd/api/logging/logging.go
+++ b/cmd/api/logging/logging.go
@@ -154,7 +154,8 @@ func LogRetrieval() gin.HandlerFunc {
 		}
 
 		if !logsReturned {
-			abort.Abort(c, fmt.Errorf("no parsed logs for the json path: %s", jsonPathParser.String()),
+			abort.Abort(c, fmt.Errorf("no parsed logs for the json path: %s from logs in %s",
+				jsonPathParser.String(), logUrl),
 				http.StatusNotFound)
 		}
 	}

--- a/hack/quick-install.sh
+++ b/hack/quick-install.sh
@@ -16,6 +16,6 @@ export KNATIVE_EVENTING_VERSION=v1.15.0
 kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/${CERT_MANAGER_VERSION}/cert-manager.yaml
 kubectl apply -f https://github.com/knative/eventing/releases/download/knative-${KNATIVE_EVENTING_VERSION}/eventing.yaml
 kubectl rollout status deployment --namespace=cert-manager --timeout=30s
-kubectl rollout status deployment --namespace=knative-eventing --timeout=30s
+kubectl rollout status deployment --namespace=knative-eventing --timeout=50s
 
 bash hack/kubearchive-install.sh

--- a/test/integration/integration_test.go
+++ b/test/integration/integration_test.go
@@ -214,9 +214,18 @@ func TestArchiveAndRead(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	portForward, errPortForward := test.PortForward([]string{"8081:8081"}, pods.Items[0].Name, "kubearchive")
-	if errPortForward != nil {
-		t.Fatal(errPortForward)
+	var portForward chan struct{}
+	var errPortForward error
+	retryErr := retry.Do(func() error {
+		portForward, errPortForward = test.PortForward([]string{"8081:8081"}, pods.Items[0].Name, "kubearchive")
+		if errPortForward != nil {
+			return errPortForward
+		}
+		return nil
+	}, retry.Attempts(3))
+
+	if retryErr != nil {
+		t.Fatal(retryErr)
 	}
 
 	defer close(portForward)
@@ -309,7 +318,7 @@ func TestArchiveAndRead(t *testing.T) {
 	}
 	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.Status.Token))
 
-	retryErr := retry.Do(func() error {
+	retryErr = retry.Do(func() error {
 		response, errResp := clientHTTP.Do(request)
 		if errResp != nil {
 			fmt.Printf("Couldn't get a response HTTP, %s\n", errResp)
@@ -513,8 +522,18 @@ func TestPagination(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	portForward, errPortForward := test.PortForward([]string{"8081"}, pods.Items[0].Name, "kubearchive")
-	if errPortForward != nil {
+
+	var portForward chan struct{}
+	var errPortForward error
+	retryErr := retry.Do(func() error {
+		portForward, errPortForward = test.PortForward([]string{"8081:8081"}, pods.Items[0].Name, "kubearchive")
+		if errPortForward != nil {
+			return errPortForward
+		}
+		return nil
+	}, retry.Attempts(3))
+
+	if retryErr != nil {
 		t.Fatal(errPortForward)
 	}
 	defer close(portForward)
@@ -530,7 +549,7 @@ func TestPagination(t *testing.T) {
 			return nil
 		}
 		return errors.New("could not retrieve Pods from the API")
-	}, retry.Attempts(30), retry.MaxDelay(4*time.Second))
+	}, retry.Attempts(60), retry.MaxDelay(4*time.Second))
 
 	if err != nil {
 		t.Fatal(err)

